### DISCUSSION TITLE
Vert.x thread blocked (only on pipeline)

### DISCUSSION THREAD

**Discussion by guillaume-gorret** [quarkusio/quarkus#32982] [Q&A] [answered]
Hello there!

I'm trying to update to Quarkus 3. And I'm facing an issue with a test.

The following test is working on my computer but failing on pipeline:

```java
@Test
@RunOnVertxContext
void testCreatePilotShouldReturn201(UniAsserter asserter) {
    asserter.execute(() -> {
        var dto = CreatePilotDTO.builder()
            .email("new@email.com")
            .lastName("Doe").firstName("John")
            .build();

        RestAssured.given()
                .contentType(ContentType.JSON)
                .body(dto)
                .put("/api/v1/users/pilots")
                .then()
                .statusCode(201);
    }).assertTrue(() -> identityRepository.existsByEmail("new@email.com"));
}
```

The error when launching on pipeline is on the line:
`.put("/api/v1/users/pilots")`

And the error is:
`java.net.SocketTimeoutException: Read timed out` and I have also `Suppressed: io.vertx.core.VertxException: Thread blocked`

Maybe the reason is that I'm trying to test every layer (rest, service, repository) on the same test ?

Thanks for your help :)

**Comment by manusa** [comment, marked as answer]
I've asked this question on [Zulip](https://quarkusio.zulipchat.com/) because I'm affected by a similar situation (combining Panache access and `RestAssured` in the same test).

@mkouba pointed out that `io.quarkus.vertx.VertxContextSupport.subscribeAndAwait(Supplier<Uni<T>>)` can be used in a blocking thread to run a non-blocking piece of code with a Vertx context. 

So I think that your test could be rewritten as:
```java
@Test
void testCreatePilotShouldReturn201() {
  var dto = CreatePilotDTO.builder()
    .email("new@email.com")
    .lastName("Doe").firstName("John")
    .build();

  RestAssured.given()
    .contentType(ContentType.JSON)
    .body(dto)
    .put("/api/v1/users/pilots")
    .then()
    .statusCode(201);
  assertTrue(VertxContextSupport.subscribeAndAwait(() -> Panache.withSession(() -> identityRepository.existsByEmail("new@email.com"))));
}
```

This approach worked for me (no need for `@RunOnVertxContext` or `UniAsserter`).

**Reply by guillaume-gorret** [reply]
That's perfect ! Thanks !

**Reply by mkouba** [reply]
FTR `VertxContextSupport.subscribeAndAwait()` is documented here: https://quarkus.io/guides/vertx#executing-asynchronous-code-from-a-blocking-thread
